Tests/Unit/Hooks/HookProviderTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+declare(strict_types = 1);
+
+namespace OliverKlee\Seminars\Tests\Unit\Hooks;
+
+use Nimut\TestingFramework\TestCase\UnitTestCase;
+use OliverKlee\Seminars\Hooks\HookProvider;
+use OliverKlee\Seminars\Interfaces\Hook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestHookInterface;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestHookImplementor;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Test case.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookProviderTest extends UnitTestCase
+{
+    /**
+     * @var array
+     */
+    private $extConfBackup = [];
+
+    protected function setUp()
+    {
+        $this->extConfBackup = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'];
+    }
+
+    protected function tearDown()
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'] = $this->extConfBackup;
+    }
+
+    /*
+     * Utility functions
+     */
+
+    /**
+     * Creates a TestHook implementor object.
+     *
+     * @return TestHookImplementor
+     */
+    private function createTestHookImplementor(): TestHookImplementor
+    {
+        return GeneralUtility::makeInstance(TestHookImplementor::class);
```

Comment:
Please use `new` in the tests unless we require some particular functionality of `makeInstance` (like singletons or XClasses).